packages/medusa/src/strategies/batch-jobs/product/import.ts

```diff
@@ -255,6 +280,47 @@ class ProductImportStrategy extends AbstractBatchJobStrategy {
     })
   }
 
+  /**
+   * Create, or retrieve by name, sales channels from the input data.
+   *
+   * NOTE: Sales channel names provided in the CSV must exist in the DB.
+   *       New sales channels will not be created.
+   *
+   * @param data an array of sales channels partials
+   * @return an array of sales channels created or retrieved by name
+   */
+  private async processSalesChannels(
+    data: Pick<SalesChannel, "name" | "id">[]
+  ): Promise<SalesChannel[]> {
+    const transactionManager = this.transactionManager_ ?? this.manager_
+    const salesChannelServiceTx =
+      this.salesChannelService_.withTransaction(transactionManager)
+
+    const salesChannels: SalesChannel[] = []
+
+    for (const input of data) {
+      let channel: SalesChannel | null = null
+
+      if (input.id) {
+        try {
+          channel = await salesChannelServiceTx.retrieve(input.id)
```

Comment:
suggestion: we could reduce the amount of data to store:

```suggestion
          channel = await salesChannelServiceTx.retrieve(input.id, { select: ["id"] })
```